interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -142,17 +150,27 @@ object TapirAdapter {
         val code           =
           response.errors.collectFirst { case HttpRequestMethod.MutationOverGetError => StatusCode.BadRequest }
             .getOrElse(StatusCode.Ok)
-        val cacheDirective = HttpUtils.computeCacheDirective(response.extensions)
-        (
-          MediaType.ApplicationJson,
-          code,
-          cacheDirective,
-          encodeSingleResponse(
-            resp,
-            keepDataOnErrors = true,
-            excludeExtensions = cacheDirective.map(_ => Set(Caching.DirectiveName))
-          )
-        )
+        val cacheDirective = computeCacheDirective(response.extensions)
+        isSSE match {
+          case true  =>
+            (
+              MediaType.TextEventStream,
+              code,
+              cacheDirective,
+              encodeTextEventStreamResponse(resp)
+            )
+          case false =>
+            (
+              MediaType.ApplicationJson,
+              code,
+              cacheDirective,
+              encodeSingleResponse(
+                resp,
+                keepDataOnErrors = true,
+                excludeExtensions = cacheDirective.map(_ => Set(Caching.DirectiveName))
+              )
+            )
+        }
```

Comment:
Couple of questions / comments here (@paulpdaniels let me know what you think as well)

1. I think the `isSSE` condition can be included in the pattern match above
2. According to the spec, SSE can also be used for `@defer` responses (which we currently support).
3. I think we can pass `None` as the `cacheDirective` for SSE responses